src/plugins/data/server/index_patterns/routes/runtime_fields/get_runtime_field.ts

```diff
@@ -46,20 +47,36 @@ export const registerGetRuntimeFieldRoute = (
 
       const indexPattern = await indexPatternsService.get(id);
 
-      const field = indexPattern.fields.getByName(name);
+      const runtimeField = indexPattern.getRuntimeField(name);
 
-      if (!field) {
+      if (!runtimeField) {
         throw new ErrorIndexPatternFieldNotFound(id, name);
       }
 
-      if (!field.runtimeField) {
-        throw new Error('Only runtime fields can be retrieved.');
+      // Access the data view fields created for the runtime field
+      let dataViewFields: DataViewField[];
+
+      if (runtimeField.type === 'composite') {
+        // For "composite" runtime fields we need to look at the "fields"
+        dataViewFields = Object.keys(runtimeField.fields!)
+          .map((subFieldName) => {
+            const fullName = `${name}.${subFieldName}`;
+            return indexPattern.fields.getByName(fullName);
+          })
+          .filter(Boolean) as DataViewField[];
+      } else {
+        dataViewFields = [indexPattern.fields.getByName(name)].filter(Boolean) as DataViewField[];
       }
 
       return res.ok({
         body: {
-          field: field.toSpec(),
-          runtimeField: indexPattern.getRuntimeField(name),
+          // New API for 7.16 & 8.x. Return an Array of DataViewFields for the runtime field
+          fields: dataViewFields,
+          // @deprecated
+          // To avoid creating a breaking change in 7.16 we continue to support
+          // the old "field" in the response
+          field: dataViewFields[0].toSpec(),
```

Comment:
I'm tempted to remove support for this. Its only been in 7.14 and 7.15. I'll address this separately, no reason to hold up this PR.

Reply:
@mattkime - if you want to make a breaking change in 8.0, even if it's only been in 7.14 and 7.15, please reach out. With the new Make it Minor changes, we have been asked to have all breaking changes approved at team/tech lead level. REST API breaking changes are included in that (but not plugin API changes).

Reply:
This is not only on `8.x`, the breaking change would also ship in `7.16`.